RandLAPACK/comps/rl_util.hh

```diff
@@ -513,6 +738,42 @@ bool orthogonality_check(
     return false;
 }
 
+/// Computes an L-2 norm of a given matrix using
+/// 10 steps of power iteration.
+/// Requires additional n * (n + 2) * sizeof(T) bytes of space.
+template <typename T>
+T get_2_norm(
+    int64_t m,
+    int64_t n,
+    T* A_dat,
+    int p,
+    RandBLAS::base::RNGState<r123::Philox4x32> state
+) {
+
+    std::vector<T> buf (n, 0.0);
+    std::vector<T> buf1 (n, 0.0);
+    std::vector<T> ATA (n * n, 0.0);
+
+    RandBLAS::dense::DenseDist DV{.n_rows = n, .n_cols = 1};
+    state = RandBLAS::dense::fill_buff(buf.data(), DV, state);
+
+    // Compute A^T*A
+    blas::syrk(Layout::ColMajor, Uplo::Lower, Op::Trans, n, m, 1.0, A_dat, m, 0.0, ATA.data(), n);
+    // Fill the upper triangular part of A^T*A
+    for(int i = 1; i < n; ++i)
+        blas::copy(n - i, ATA.data() + i + ((i-1) * n), 1, ATA.data() + (i - 1) + (i * n), n);
+
+    T prev_norm_inv = 1.0;
+    for(int i = 0; i < p; ++i)
+    {
+        gemv(Layout::ColMajor, Op::NoTrans, n, n, prev_norm_inv, ATA.data(), n, buf.data(), 1, 0.0, buf1.data(), 1);
```

Comment:
Less important note: don't use ``gemv`` if you know your matrix is symmetric or Hermitian. You should use ``symv`` or ``hemv`` in these settings.

Reply:
This comment isn't super relevant though because I want you to delete this code and not form the gram matrix explicitly.

Reply:
Understood!